generators/entity-server/templates/src/main/resources/config/liquibase/changelog/added_entity.xml.ejs

```diff
@@ -217,7 +217,11 @@
             <%_ } _%>
             <%_ for (idx in relationships) {
                     let loadColumnType = 'numeric';
-                    if (relationships[idx].relationshipValidate === true && relationships[idx].relationshipRequired) {
+                    if (relationships[idx].relationshipValidate === true && relationships[idx].relationshipRequired
+                        && (relationshipType === "many-to-one"
+                            || (relationshipType === "one-to-one" && relationships[idx].ownerSide === true
+                                && (relationships[idx].useJPADerivedIdentifier == null || relationships[idx].useJPADerivedIdentifier === false))
+                    )) {
```

Comment:
This now matches https://github.com/jhipster/generator-jhipster/blob/706fc1f91d69de989ea7df3f49586cabdf1b6de8/generators/entity-server/templates/src/main/resources/config/liquibase/fake-data/table.csv.ejs#L203-L207